chapter_4/ex05_config_file_package/CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+# CMakeLists file for the Chapter 4 Example nr. 05: 
+# Config-file package
+#
+# SPDX-License-Identifier: MIT
+
+project(
+  ch4_ex05_lib
+  VERSION 1.0
+  DESCRIPTION "Chapter 4 Example 05, Config-file package"
+  LANGUAGES CXX)
+
+
+include(GNUInstallDirs) # Defines the ${CMAKE_INSTALL_INCLUDEDIR} variable.
+
+# This is the default install directory for config-file package cmake files
+# It is relative to install prefix.
+set(CFP_INSTALL_DIR cmake CACHE PATH "Installation directory for config-file package cmake files")
+
+# Define a static library target named `ch4_ex05_lib` 
+add_library(ch4_ex05_lib STATIC)
+
+# Specify source files for target named `ch4_ex05_lib`
+target_sources(ch4_ex05_lib PRIVATE src/lib.cpp)
+
+# Specify the include directories for the target named `ch4_ex05_lib`
+target_include_directories(ch4_ex05_lib PUBLIC 
+          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+          $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
```

Comment:
Please avoid using the `INSTALL_INTERFACE` genex. The way most people use it requires intermingling unrelated build and usage requirement declarations with code that is only concerned with install rules. Using the `INCLUDES DESTINATION` form of `install(TARGETS)` helps with keeping code that only affects install rules with other commands that do as such.

Reply:
Alright, that makes sense. Will do.